packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -97,6 +97,17 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
         this.writeToUserHome({ recentRoots });
     }
 
+    async removeRecentWorkspace(uri: string): Promise<void> {
+        const recentRoots = await this.getRecentWorkspaces();
+        const index = recentRoots.indexOf(uri);
+        if (index !== -1) {
+            recentRoots.splice(index, 1);
+        }
+        this.writeToUserHome({
+            recentRoots
+        });
```

Comment:
Shouldn't we only write to the user home if the `index !== -1`?

Reply:
Done